src/thor/trace_route_action.cc

```diff
@@ -437,7 +437,10 @@ thor_worker_t::map_match(Api& request) {
 
       // The following logic put break points (matches results) on edge candidates to form legs
       // logic assumes the both match results and edge candidates are topologically sorted in correct
-      // order
+      // order. Only the first location will be populated with corresponding input date_time
+      std::string date_time = options.mutable_shape(0)->date_time();
```

Comment:
this isnt gauranteed to be populated when there isnt a time dependency, you have to check `has_date_time` first and set to empty string otherwise

Reply:
i dont understand how the tests dont throw on this

Reply:
good catch. Maybe when it is not populated, it returns "" instead throwing.  Should check before use.